### ISSUE TITLE
Embedded sdk didn't support new parse server.

### ISSUE THREAD

**Issue by evileye2000** [ParsePlatform/parse-embedded-sdks#63]
Parse server needs to api for change server. But, embedded sdk did not support yet. 
And, parse server did not support push too.

I want to know the schedule about update sdk and push support.

Thanks

**Comment by evileye2000** [comment]
I sent a new patch for connect parse server.

**Comment by nlutsenko** [comment]
Yup, looks like it, thanks for reporting.
We are going to use this issue to track progress and impementation of this.
Here is the brief list of all things that needs to be addressed so all implementation of embedded Parse SDK support Open Parse Server:
- [ ] Add `parseInitializeWithServerURL()` to `include/parse.h`
- [ ] Add `parseInitializeWithServerURL()` implementation to `unix/src/parse.c`
- [ ] Add `parseInitializeWithServerURL()` implementation to `rtos/parse_impl.c`
- [ ] Add `serverURL` support to `unix/yun/yun.c`
- [ ] Add `beginWithServerURL()` to Arduino SDK in `src/internal/ParseClient.h`
- [ ] Add `beginWithServerURL()` implementation to Arduino in `src/internal/esp8266/ParseClient.cpp`
- [ ] Add `beginWithServerURL()` implementation to Arduino in `src/internal/yun/ParseClient.cpp`
- [ ] Add `beginWithServerURL()` implementation to Arduino in `src/internal/zero/ParseClient.cpp`

Feel free to jump on any one of these and link back to this issue.

_This comment is going to be updating by itself as progress is made._

**Comment by hramos** [comment]
There's one more thing that needs to be addressed. The client key (or REST API Key, in this case) is optional when Parse Server is used. The SDKs should provide an initialization method that only requires an application id.

**Comment by notiles** [comment]
Hello, Thanks a lot for that, have you got any update on It? Do you schedule a fix soon ?

**Comment by nlutsenko** [comment]
Hey @notiles, not update on it yet, but I'll do my best to review any contributions that implement this.

**Comment by evileye2000** [comment]
@hramos I sent a new PR. we can assign "NULL" to client key like android api for parse server.

**Comment by evileye2000** [comment]
@notiles Now, embedded c can connect parse server. But arduino is not yet patched.

**Comment by notiles** [comment]
@evileye2000 Thanks a lot but my biggest priority is on arduino YUN. I read that Parse use the PPNS protocol, if we patch the Arduino Library, does we have to configure our server with those specifications? https://github.com/ParsePlatform/parse-server/wiki/PPNS-Protocol-Specification

**Comment by hramos** [comment]
@notiles yes, you will need to set up a PPNS server to continue sending notifications to Parse IoT devices.

**Comment by notiles** [comment]
Thanks @hramos, what I understand about PPNS is that I have to setup an ssl certificate on my server, but is there anything else to setup on the server or the parse server sdk do the job himself?
